docs/addons/gmaestro.md

```diff
@@ -0,0 +1,114 @@
+   1. Deployment section `MAESTRO_CLIENT_ID`
+   2. ConfigMap section `prometheus.configs.remote_write.basic_auth.password`.
+   3. ConfigMap section `loki.configs.clients.basic_auth.password`
+
+## Installation
+
+Using [npm](https://npmjs.org):
+
+```bash
+$ npm i @granulate/gmaestro-eks-blueprints-addon
+```
+
+## Usage
+
+```typescript
+import 'source-map-support/register';
+import * as CDK from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import {GmaestroAddOn} from '@granulate/gmaestro-eks-blueprints-addon';
+
+const app = new cdk.App();
+
+const addOn = new GmaestroAddOn({
+        clientIdSecretName: "<client id secret name>", // Create and copy from gMaestro deployment yaml
+        clientName: "<client name>",
+        clusterName: "<cluster name>",
+        namespace: "<namespace>",
+        grafanaMetricsSecretName: "<grafana metrics secret name>", // Create and copy from gMaestro deployment yaml
+        grafanaLogsSecretName: "<grafana logs secret name>", // Create and copy from gMaestro deployment yaml
+    });
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+
+Use the following command to validate that gMaestro installed successfully:
+
+```bash
+$ kubectl get pods --all-namespaces | grep granulate-maestro
+
+NAMESPACE     NAME                                 READY   STATUS    RESTARTS   AGE
+default       granulate-maestro-6947dc87bc-k5nfc   2/2     Running   0          11m
+kube-system   aws-node-9rhgx                       1/1     Running   0          16m
+kube-system   coredns-d5b9bfc4-8v8k5               1/1     Running   0          21m
+kube-system   coredns-d5b9bfc4-r5sdb               1/1     Running   0          21m
+kube-system   kube-proxy-js5pn                     1/1     Running   0          16m
+```
+
+After a few seconds, you will gain full visibility into your K8s cluster objects.
+The first rightsizing recommendations may take up to 5 minutes to load.
+
+## `gMaestroAddOn` Required (props)
+Take the following parameter from the sample YAML file that was downloaded.
+
+#### `clientIdSecretName: string`
+
+The secret name from the Prerequisite section 3.i.
+
+#### `clientName: string`
+
+Copy from the ConfigMap section `prometheus.configs.scrape_configs.static_configs.labels.client_name` value
+
+#### `clusterName: string`
+
+Copy from the Deployment section `MAESTRO_SERVICE_NAME` value
+
+#### `namespace: string` (optional)
+
+The namespace where gMaestro will be installed. `default` namespace is used as default.
+
+#### `grafanaMetricsSecretName: string`
+
+The secret name from the Prerequisite section 3.ii.
+
+#### `grafanaLogsSecretName: string`
+
+The secret name from the Prerequisite section 3.iii.
+
+
+## Security issue
+
+As the implementation uses getSecretValue(), the secret parameters will be displayed as plain text in the CloudFormation. These secrets are gMaestro’s secrets which don’t affect the user.
```

Comment:
Many customers have no idea what `getSecretValue()` means in this context.

Let's state the following:

1. The implementation requires access to the AWS Secrets Manager at build time to retrieve secret values.
2. The secret values will be stored as plain text in the resulting CloudFormation stack, meaning that any user with access to view CloudFormation stack can gain access to these secrets.

Note: These secrets are specific to gMaestro and don't affect customer account beyond the scope of the gMaestro add-on.

My wording can surely be improved further, I was just attempting to convey the point, please feel free to adjust.